server/src/main/java/io/crate/execution/ddl/tables/AddColumnRequest.java

```diff
@@ -61,32 +61,50 @@ public AddColumnRequest(@Nonnull RelationName relationName,
 
     public AddColumnRequest(StreamInput in) throws IOException {
         super(in);
-        relationName = new RelationName(in);
+        AddColumnRequest request = readColsToAdd(in);
+        this.relationName = request.relationName();
+        this.colsToAdd = request.references();
+        this.checkConstraints = request.checkConstraints();
+        this.pKeyIndices = request.pKeyIndices();
+    }
+
+    public static AddColumnRequest readColsToAdd(StreamInput in) throws IOException {
+        final RelationName relationName = new RelationName(in);
```

Comment:
I had an idea to introduce a record but then I thought that it's not much different from the very first iteration when we just inlined streamable `AddColumnRequest` into `CreateTableRequest`. 

But yes, re-used in Add/Create record is probably better than inlining a request. Will do.

Reply:
Thx, implemented in https://github.com/crate/crate/pull/13934/commits/ad11a97252031faef3f87c2b29c2cc9ba0d2c226

We avoid inlining and reuse of streaming looks more natural.

Reply:
UPD: It breaks type param bounds in one of Transport Actions. 

After https://github.com/crate/crate/commit/ad11a97252031faef3f87c2b29c2cc9ba0d2c226 `TransportCreateTableAction` was broken with 

`Type parameter 'io.crate.execution.ddl.tables.CreateTableRequest' is not within its bound; should extend 'org.elasticsearch.action.support.master.MasterNodeRequest<io.crate.execution.ddl.tables.CreateTableRequest>'`

and after attempted fix https://github.com/crate/crate/pull/13934/commits/b6c43152f675bfd6dbd696c4dbb7a5becbb944ed
`TransportAddColumnAction` was broken with as `AbstractDDLTransportAction` expects 
`Request extends AcknowledgedRequest<Request>`

Reply:
Created a record in https://github.com/crate/crate/pull/13934/commits/787121894efbcd36f9658f355468e6a90427bf29

So of read we use record, for writes still common static method to avoid creation of `ReferencesAndConstraints` only to call its `writeTo` method.